documentation/modules/ROOT/pages/connectors/db2.adoc

```diff
@@ -0,0 +1,1503 @@
+
+Each snapshot consists of the following steps:
+
+1. Determine the tables to be snapshoted from whitelist/blacklist
+2. Obtain a lock on each of the monitored tables to ensure that no structural changes can occur to any of the tables.
+The level of the lock is determined by `snapshot.isolation.mode` configuration option.
+3. Read the maximum LSN ("log sequence number") position in the server's transaction log.
+4. Capture the structure of all relevant tables.
+5. Optionally release the locks obtained in step 2, i.e. the locks are held usually only for a short period of time.
+6. Scan all of the relevant database tables and schemas as valid at the LSN position read in step 3, and generate a `READ` event for each row.
+Then write that event to the appropriate table-specific Kafka topic with the maxiumun LSN, i.e. all snapshot insert operations (i.e those with opcode='r') have the same LSN. 
+7. Record the successful completion of the snapshot in the connector offsets.
+
+=== Reading the change data tables
+
+Upon first start-up, the connector takes a structural snapshot of the structure of the requested tables
+and persists this information in its internal database history topic.
+Then the connector identifies a change table for each of the source tables and executes the main loop
+
+1. For each change table read all changes that were created between last stored maximum LSN and current maximum LSN
+2. Order the read changes incrementally according to commit LSN and change LSN.
+This assures that the changes are replayed by Debezium in the same order as were made to the database.
+3. Pass commit and change LSNs as offsets to Kafka Connect.
+4. Store the maximum LSN and repeat the loop.
+
+After a restart, the connector will resume from the offset (commit and change LSNs) where it left off before.
+
+The connector is able to detect whether the CDC is enabled or disabled for whitelisted source table during the runtime and modify its behaviour.
+
+=== Topic names
+
+The Db2 connector writes events for all insert, update, and delete operations on a single table to a single Kafka topic. The name of the Kafka topics always takes the form _databaseName_._schemaName_._tableName_, where _databaseName_ is the logical name of the connector as specified with the `database.server.name` configuration property, _schemaName_ is the name of the schema where the operation occurred, and _tableName_ is the name of the database table on which the operation occurred.
+
+Unlike for SQL Server, it is only possible for a connector to streams changes from one Db2 database.
+
+For example, consider a Db2 installation with an `mydatabase` database that contains four tables: `products`, `products_on_hand`, `customers`, and `orders` in schema `myschema` then the connector would produce events on these four Kafka topics:
+
+* `mydatabase.myschema.products`
```

Comment:
Schema and table name should be upper-cased as per the recent discussion with @lga-zurich.

Reply:
Done